tests/vm_test.go

```diff
@@ -971,309 +969,149 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
 
-			Expect(strings.Contains(stdErr, "Error from server (AlreadyExists): error when creating")).To(BeTrue(), "command should error when creating VM second time")
+			_, err = virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
+			Expect(err).To(MatchError(errors.IsNotFound, "k8serrors.IsNotFound"))
 		})
 
-		DescribeTable("[release-blocker][test_id:299]should create VM via command line using all supported API versions", decorators.Conformance, func(version string) {
-			vmi := libvmifact.NewAlpine()
-			vm := libvmi.NewVirtualMachine(vmi, libvmi.WithRunStrategy(v1.RunStrategyAlways))
-			vm.Namespace = testsuite.GetTestNamespace(vm)
-			vm.APIVersion = version
-
-			vmJson, err := tests.GenerateVMJson(vm, workDir)
-			Expect(err).ToNot(HaveOccurred(), "Cannot generate VMs manifest")
-
-			By("Creating VM using k8s client binary")
-			_, _, err = clientcmd.RunCommand(testsuite.GetTestNamespace(nil), k8sClient, "create", "-f", vmJson)
+		DescribeTable("when stopping a VM", func(gracePeriod *int64) {
+			By("Creating VM")
+			vm := libvmi.NewVirtualMachine(libvmifact.NewAlpine())
+			vm, err := virtClient.VirtualMachine(testsuite.GetTestNamespace(vm)).Create(context.Background(), vm, metav1.CreateOptions{})
 			Expect(err).ToNot(HaveOccurred())
 
 			By("Waiting for VMI to start")
 			Eventually(ThisVMIWith(vm.Namespace, vm.Name), 120*time.Second, 1*time.Second).Should(BeRunning())
 
-			By("Listing running pods")
-			stdout, _, err := clientcmd.RunCommand(testsuite.GetTestNamespace(nil), k8sClient, "get", "pods")
+			By("Getting current vmi instance")
+			originalVMI, err := virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
 			Expect(err).ToNot(HaveOccurred())
 
-			By("Ensuring pod is running")
-			expectedPodName := getExpectedPodName(vm)
-			podRunningRe, err := regexp.Compile(fmt.Sprintf("%s.*Running", expectedPodName))
+			By("Getting current vm instance")
+			originalVM, err := virtClient.VirtualMachine(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
 			Expect(err).ToNot(HaveOccurred())
 
-			Expect(podRunningRe.FindString(stdout)).ToNot(Equal(""), "Pod is not Running")
-
-			By("Checking that VM is running")
-			stdout, _, err = clientcmd.RunCommand(testsuite.GetTestNamespace(nil), k8sClient, "describe", "vmis", vm.GetName())
+			By("Performing dry run stop")
+			options := &v1.StopOptions{DryRun: []string{metav1.DryRunAll}}
+			if gracePeriod != nil {
+				options.GracePeriod = gracePeriod
+			}
+			err = virtClient.VirtualMachine(vm.Namespace).Stop(context.Background(), vm.Name, options)
 			Expect(err).ToNot(HaveOccurred())
 
-			Expect(vmRunningRe.FindString(stdout)).ToNot(Equal(""), "VMI is not Running")
-		},
-			Entry("with v1 api", "kubevirt.io/v1"),
-			Entry("with v1alpha3 api", "kubevirt.io/v1alpha3"),
-		)
+			By("Checking that VM is still running")
+			Expect(ThisVM(vm)).To(BeRunning())
```

Comment:
Same, this should be in a `Consistently`

Reply:
Please see this https://github.com/kubevirt/kubevirt/pull/13029#discussion_r1793298634